starter_code/src/movies.js

```diff
@@ -2,14 +2,103 @@
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
 
+const orderByYear = function(arr) {
+    let sortYears = arr.slice().sort(function(a,b) {
+        if (a.year > b.year) return 1
+        if (a.year < b.year) return -1
+  
+        if (a.title > b.title) return 1
+        if (a.title < b.title) return -1
+    })
+    return sortYears
+  }
+  
+console.log(orderByYear(movies));
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+const howManyMovies = function(arr) {
+    return arr.filter(movie => movie.director == "Steven Spielberg" && movie.genre.indexOf('Drama') != -1).length;
+}
+
+console.log(howManyMovies(movies))
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+const orderAlphabetically = function(arr) {
+    let arrTwo = JSON.parse(JSON.stringify(arr))
+    arrTwo.sort(function(a,b) {
+        if (a.title > b.title) return 1
+        if (a.title < b.title) return -1
+    })
+
+    let newArr = [];
```

Comment:
Another valid approach is let newArr = arrTwo.map(eachMovie => eachMovie.title).slice(0, 20);

Since we know there are more than 20 movies inside data.js it's ok to do it that way as well.